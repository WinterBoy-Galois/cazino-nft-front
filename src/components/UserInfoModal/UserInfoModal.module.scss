@import '../../styles/shared';

.details {
  border-radius: 8px;
  background-color: $user-info-main-blue;
  padding: 1rem;

  @include media-breakpoint-up(md) {
    width: 80%;
    margin: 0 auto;
  }

  &__item {
    display: flex;
    width: 100%;
    margin-top: 1rem;

    &:first-of-type {
      margin: 0;
    }

    &__label {
      width: 50%;
      color: $user-info-label-blue;
      text-transform: uppercase;

      @include media-breakpoint-up(md) {
        text-align: right;
        margin-right: 1rem;
      }
    }

    &__value {
      display: flex;
      justify-content: flex-end;
      align-items: center;
      width: 50%;
      font-weight: 300;

      @include media-breakpoint-up(md) {
        justify-content: flex-start;
      }
    }
  }
}

.username {
  margin: 1rem 0 2rem;

  @include media-breakpoint-up(md) {
    width: 100%;
    justify-content: center;
    margin: 0 0 1rem;
  }

  &--mobile {
    @include media-breakpoint-up(md) {
      display: none !important;
    }
  }

  &--desktop {
    display: none !important;

    @include media-breakpoint-up(md) {
      display: flex !important;
    }
  }

  &__hidden {
    font-weight: 600;
    color: #56718d;
  }
}

.game-icon {
  width: 14px;
  height: 14px;
  fill: $white;
  vertical-align: middle;
  margin-right: 5px;

  &__inner {
    fill: $user-info-main-blue;
  }
}

.button {
  width: 66%;
  max-width: 220px;
  margin: 3rem auto 0;

  @include media-breakpoint-up(md) {
    margin: 2rem auto 0.5rem;
  }

  &__spacer {
    height: 40px;
  }
}
