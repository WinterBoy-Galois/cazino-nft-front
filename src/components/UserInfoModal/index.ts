export { default } from './UserInfoModal';
export { UserInfoModalWithData } from './UserInfoModal';
