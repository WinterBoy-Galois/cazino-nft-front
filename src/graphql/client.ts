import { ApolloClient, InMemoryCache, from, split } from '@apollo/client';
import { WebSocketLink } from '@apollo/client/link/ws';
import { TokenRefreshLink } from 'apollo-link-token-refresh';
import possibleTypes from './possibleTypes.json';
import { getMainDefinition } from '@apollo/client/utilities';
import { HttpLink, defaultDataIdFromObject } from '@apollo/client';
import jwtDecode from 'jwt-decode';
import { getEpoch } from '../common/util/date.util';
import { setContext } from '@apollo/client/link/context';
import { AuthType } from '../state/models/auth.model';
import { appConfig } from '../common/config';

const cache = new InMemoryCache({
  dataIdFromObject(responseObject) {
    switch (responseObject.__typename) {
      default:
        return defaultDataIdFromObject(responseObject);
    }
  },
  addTypename: true,
  possibleTypes,
});

let wsLink: WebSocketLink;

const getApolloClient = (
  onAccessTokenRefresh: (accessToken: string) => void,
  onSignOut: () => void,
  authType: AuthType,
  accessToken?: string
) => {
  cache.reset();

  if (wsLink) {
    (wsLink as any).subscriptionClient.close();
  }

  wsLink = new WebSocketLink({
    uri: `${appConfig.apiBasePathWS}/graphql`,
    options: {
      reconnect: true,
      connectionParams: {
        authToken: accessToken ?? undefined,
      },
    },
  });

  const httpLink = new HttpLink({
    uri: `${appConfig.apiBasePath}/graphql`,
    credentials: 'include',
  });

  const authLink = setContext((_, { headers }) => {
    if (accessToken) {
      return {
        headers: {
          ...headers,
          authorization: `Bearer ${accessToken}`,
        },
      };
    }

    return headers;
  });

  const tokenLink = new TokenRefreshLink({
    accessTokenField: 'accessToken',
    isTokenValidOrUndefined: () => {
      if (accessToken) {
        const { exp } = jwtDecode(accessToken);

        if (exp < getEpoch()) {
          return false;
        }
      } else if (authType === 'SIGNED_IN') {
        return false;
      }

      return true;
    },
    fetchAccessToken: () => {
      return fetch(`${appConfig.apiBasePath}/refresh_tokens`, {
        method: 'POST',
        credentials: 'include',
      });
    },
    handleFetch: accessToken => onAccessTokenRefresh(accessToken),
    handleError: () => onSignOut(),
  });

  const link = split(
    ({ query }) => {
      const definition = getMainDefinition(query);
      return definition.kind === 'OperationDefinition' && definition.operation === 'subscription';
    },
    wsLink,
    authLink.concat(httpLink)
  );

  return new ApolloClient({
    cache,
    link: from([tokenLink, link]),
  });
};

export default getApolloClient;
