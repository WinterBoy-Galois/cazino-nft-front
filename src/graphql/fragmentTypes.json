{
  "__schema": {
    "types": [
      {
        "kind": "UNION",
        "name": "PublicUserOrError",
        "possibleTypes": [{ "name": "PublicUser" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "LoginResultOrError",
        "possibleTypes": [{ "name": "LoginResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "UserOrError",
        "possibleTypes": [{ "name": "User" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "SignedInTokenOrError",
        "possibleTypes": [{ "name": "LoginResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "BetResultOrError",
        "possibleTypes": [{ "name": "BetResult" }, { "name": "GenericErrorArray" }]
      },
      {
        "kind": "UNION",
        "name": "BooleanOrError",
        "possibleTypes": [{ "name": "GenericBoolean" }, { "name": "GenericErrorArray" }]
      }
    ]
  }
}
