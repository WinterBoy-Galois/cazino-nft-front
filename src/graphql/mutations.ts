import gql from 'graphql-tag';

export const CHANGE_SERVER_SEED = gql`
  mutation ChangeServerSeed {
    changeServerSeed {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ACTIVATE_ACCOUNT = gql`
  mutation ActivateAccount($code: String!) {
    activateAccount(code: $code) {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_IN = gql`
  mutation SignIn($email: String, $password: String, $remember: Boolean) {
    signIn(email: $email, password: $password, remember: $remember) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          email
          isActivated
          depositAddress
          refCode
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_UP = gql`
  mutation SignUp(
    $email: String
    $password: String
    $username: String
    $token: String
    $ref: String
  ) {
    registerUser(
      username: $username
      email: $email
      password: $password
      captcha: $token
      ref: $ref
    ) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          depositAddress
          refCode
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SIGN_OUT = gql`
  mutation SignOut {
    signOut
  }
`;

export const RESEND_ACTIVATION_CODE = gql`
  mutation ResendActivationCode {
    resendActivationCode {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const RESET_PASSWORD = gql`
  mutation ResetPassword($token: String!, $newPassword: String!) {
    resetPassword(uuid: $token, newPassword: $newPassword) {
      ... on LoginResult {
        accessToken
        user {
          id
          username
          avatarUrl
          balance
          depositAddress
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const RECOVER_PASSWORD = gql`
  mutation RecoverPassword($email: String!) {
    forgotPassword(email: $email) {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_PASSWORD = gql`
  mutation UpdatePassword($oldPassword: String!, $newPassword: String!) {
    modifyPassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_PREFERENCES = gql`
  mutation UpdatePreferences($hideUsername: Boolean!, $hideProfit: Boolean!, $hideWager: Boolean!) {
    modifyPreferences(
      hideUsername: $hideUsername
      hideTotalProfit: $hideProfit
      hideTotalWager: $hideWager
    ) {
      ... on User {
        id
        hideUsername
        hideTotalProfit
        hideTotalWager
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const UPDATE_AVATAR = gql`
  mutation updateAvatar($index: Int!) {
    modifyAvatar(index: $index) {
      ... on User {
        id
        avatarUrl
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_DICE = gql`
  mutation makeBetDice($betAmount: Float!, $over: Boolean!, $target: Float!) {
    makeBetDice(betAmount: $betAmount, over: $over, target: $target) {
      ... on BetResultDice {
        id
        result
        lucky
        multiplier
        profit
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_CLAMS = gql`
  mutation makeBetClams($betAmount: Float!, $selection: [Int]) {
    makeBetClams(betAmount: $betAmount, selection: $selection) {
      ... on BetResultClams {
        id
        result
        lucky
        multiplier
        profit
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_GOALS = gql`
  mutation makeBetGoals($betAmount: Float!, $difficulty: GoalsDifficulty!) {
    makeBetGoals(betAmount: $betAmount, difficulty: $difficulty) {
      ... on GoalsGameSetup {
        session {
          betId
          betAmount
          currentStep
          selections {
            step
            luckySpots
            selected
          }
          difficulty
          profits {
            step
            multiplier
            profit
          }
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ADVANCE_GOALS = gql`
  mutation advanceGoals($betId: ID!, $selection: Int!) {
    advanceGoals(betId: $betId, selection: $selection) {
      ... on GoalsStep {
        id
        totalProfit {
          multiplier
          profit
        }
        nextProfit {
          multiplier
          profit
        }
        nextStep
        profitCut
        allowNext
      }
      ... on GoalsComplete {
        id
        lucky
        profit {
          multiplier
          profit
        }
        balance
        profitCut
        result {
          step
          luckySpots
          selected
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CASH_OUT_GOALS = gql`
  mutation cashoutGoals($betId: ID!) {
    cashoutGoals(betId: $betId) {
      ... on GoalsComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        profitCut
        result {
          step
          luckySpots
          selected
        }
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const WITHDRAW = gql`
  mutation withdraw($amount: Float!, $address: String!) {
    withdraw(amount: $amount, address: $address) {
      ... on GenericBoolean {
        result
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CLAIM_FAUCET = gql`
  mutation claimFaucet {
    claimFaucet {
      ... on User {
        balance
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const MAKE_BET_MINES = gql`
  mutation makeBetMines($betAmount: Float!, $mines: Int!) {
    makeBetMines(betAmount: $betAmount, mines: $mines) {
      __typename
      ... on MinesGameSetup {
        session {
          ... on MinesGameSessionSetup {
            betId
            betAmount
            mines
            open
            totalProfit {
              multiplier
              profit
            }
            nextProfit {
              multiplier
              profit
            }
            allowNext
            profitCut
          }
        }
        balance
        maxProfit
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const ADVANCE_MINES = gql`
  mutation advanceMines($betId: ID!, $selection: Int!) {
    advanceMines(betId: $betId, selection: $selection) {
      ... on MinesStep {
        id
        totalProfit {
          multiplier
          profit
        }
        nextProfit {
          multiplier
          profit
        }
        profitCut
        allowNext
        result
      }
      ... on MinesComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        minePositions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CASH_OUT_MINES = gql`
  mutation cashoutMines($betId: ID!) {
    cashoutMines(betId: $betId) {
      ... on MinesComplete {
        id
        lucky
        balance
        profit {
          multiplier
          profit
        }
        minePositions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const CLAIM_BONUS = gql`
  mutation claimBonus($bonusId: ID!) {
    claimBonus(bonusId: $bonusId) {
      ... on User {
        balance
      }
      ... on GenericErrorArray {
        errors {
          code
          message
        }
      }
    }
  }
`;

export const CLAIM_COMMISSION = gql`
  mutation claimCommissions {
    claimCommissions {
      ... on User {
        id
        username
        email
        balance
        hideUsername
        hideTotalProfit
        hideTotalWager
        isActivated
        avatarUrl
        totalWager
        totalProfit
        mostPlayed
        totalBets
        luckyBets
        depositAddress
        refCode
        refCommissions
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;
