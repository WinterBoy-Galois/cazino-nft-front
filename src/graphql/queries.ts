import gql from 'graphql-tag';

export const RECENT_BETS = gql`
  {
    recentBets {
      allBets {
        id
        time
        userid
        username
        gameid
        bet
        profit
        multiplier
      }
      myBets {
        id
        time
        userid
        username
        gameid
        bet
        profit
        multiplier
      }
    }
  }
`;

export const LEADERBOARDS = gql`
  {
    leaderboards {
      daily {
        username
        userid
        wager
        bonus
      }
      weekly {
        username
        userid
        wager
        bonus
      }
      monthly {
        username
        userid
        wager
        bonus
      }
    }
  }
`;

export const USER_INFO = gql`
  query UserInfo($userId: ID) {
    userInfo(id: $userId) {
      __typename
      ... on PublicUser {
        id
        username
        avatarUrl
        totalWager
        totalProfit
        mostPlayed
        totalBets
        luckyBets
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const USER_INFO_AVATAR_URL = gql`
  query UserInfoAvatarUrl($userId: ID) {
    userInfo(id: $userId) {
      __typename
      ... on PublicUser {
        id
        avatarUrl
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const BET_DETAILS = gql`
  query BetDetails($betId: ID) {
    betDetails(id: $betId) {
      ... on BetDetails {
        bet
        profit
        profitCut
        multiplier
        gameResult {
          ... on DiceResult {
            target
            over
            winChance
            resultFloat
          }
          ... on MinesResult {
            mineCount
            minePositions
            open
          }
          ... on GoalsResult {
            difficulty
            selections {
              step
              luckySpots
              selected
            }
          }
          ... on ClamsResult {
            selection
            resultInteger
          }
        }
      }
    }
  }
`;

export const BET_DETAILS_SERVER_SEED = gql`
  query BetDetailsServerSeed($betId: ID) {
    betDetails(id: $betId) {
      ... on BetDetails {
        seedDetails {
          __typename
          ... on SeedDetailsOther {
            serverSeedHash
          }
          ... on SeedDetailsOwn {
            activeGames
            serverSeedHash
            serverSeed
            clientSeed
            nonce
            results
            verificationUrl
          }
        }
      }
    }
  }
`;

export const ME = gql`
  query Me {
    me {
      id
      username
      avatarUrl
      balance
      email
      isActivated
      depositAddress
      refCommissions
      refCode
    }
  }
`;

export const AFF_STATS = gql`
  query affStats {
    affStats {
      bets
      refs
      wager
      commissions
    }
  }
`;

export const ME_STATISTICS_PREFERENCES = gql`
  query Me {
    me {
      id
      totalWager
      totalProfit
      mostPlayed
      totalBets
      luckyBets
      hideUsername
      hideTotalProfit
      hideTotalWager
    }
  }
`;

export const SETUP_CASHIER = gql`
  query SetupCashier {
    setupCashier {
      networkFee
      depositConfirmations
      minWithdraw
    }
    me {
      id
      depositAddress
    }
  }
`;

export const DEPOSITS = gql`
  query Deposits($page: Int, $limit: Int) {
    transactionsDeposit(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        status
        time
        hash
        amount
      }
    }
  }
`;

export const SETUP_DICE = gql`
  query SetupDice {
    setupDice {
      ... on DiceGameSetup {
        minProbability
        maxProbability
        he
      }
    }
  }
`;

export const SETUP_CLAMS = gql`
  query setupClams {
    setupClams {
      ... on ClamsGameSetup {
        he
      }
    }
  }
`;

export const SETUP_GOAL = gql`
  query setupGoals {
    setupGoals {
      ... on GoalsGameSetup {
        session {
          betId
          betAmount
          currentStep
          selections {
            step
            luckySpots
            selected
          }
          difficulty
          profits {
            step
            multiplier
            profit
          }
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
        maxProfit
      }
    }
  }
`;

export const TRANSACTION_BETS = gql`
  query Bets($page: Int, $limit: Int) {
    transactionsBet(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        time
        game
        amount
        multiplier
        profit
      }
    }
  }
`;

export const TRANSACTION_AFFILIATES = gql`
  query AffiliateClaims($page: Int, $limit: Int) {
    transactionsAffiliate(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        claimedAt
        amount
      }
    }
  }
`;

export const TRANSACTION_WITHDRAWALS = gql`
  query transactionsWithdraw($page: Int, $limit: Int) {
    transactionsWithdraw(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        status
        time
        hash
        address
        amount
        fee
      }
    }
  }
`;

export const TRANSACTION_BONUSES = gql`
  query transactionsBonus($page: Int, $limit: Int) {
    transactionsBonus(page: $page, limit: $limit) {
      page
      limit
      total
      items {
        id
        givenAt
        claimedAt
        expiresAt
        amount
        position
        type
        wager
      }
    }
  }
`;

export const FAUCET_INFO = gql`
  query faucetInfo {
    faucetInfo {
      ... on FaucetInfo {
        amount
        every
        canClaim
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const SETUP_MINES = gql`
  query setupMines {
    setupMines {
      ... on MinesGameSetup {
        session {
          betId
          betAmount
          mines
          open
          totalProfit {
            multiplier
            profit
          }
          nextProfit {
            multiplier
            profit
          }
          allowNext
          profitCut
        }
        balance
        maxProfit
      }
      ... on GenericErrorArray {
        errors {
          source
          code
          message
          args
        }
      }
    }
  }
`;

export const BONUSCLAIMS = gql`
  query bonusClaims {
    bonusClaims {
      ... on BonusClaim {
        id
        givenAt
        claimedAt
        expiresAt
        amount
        position
        type
        wager
      }
    }
  }
`;
