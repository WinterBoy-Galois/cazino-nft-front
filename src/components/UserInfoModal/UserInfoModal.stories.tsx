import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, number, text, select, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import { ApolloError } from '@apollo/client';

import UserInfoModal from '.';
import { appConfig } from '../../common/config';

const mockData = {
  userInfo: {
    __typename: 'PublicUser',
    id: '1',
    username: 'AuYHKS',
    avatarUrl: `${appConfig.apiBasePath}/ava/m1.svg`,
    totalWager: 0,
    totalProfit: 0,
    mostPlayed: 'CLAMS',
    totalBets: 27,
    luckyBets: 4,
  },
};

storiesOf('Components/UserInfoModal', module)
  .addDecorator(withKnobs)
  .add('default', () => {
    return (
      <UserInfoModal
        show={true}
        data={mockData}
        loading={false}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  })
  .add('anonymous user', () => {
    return (
      <UserInfoModal
        show={true}
        data={{
          ...mockData,
          userInfo: {
            ...mockData.userInfo,
            avatarUrl: `${appConfig.apiBasePath}/ava/m1.svg`,
            username: null,
            totalWager: null,
            totalProfit: null,
          },
        }}
        // data={mockData}
        loading={false}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  })
  .add('User not found', () => {
    return (
      <UserInfoModal
        show={true}
        data={{
          userInfo: {
            __typename: 'GenericErrorArray',
            errors: [{ type: '', field: '', messageKey: '' }],
          },
        }}
        loading={false}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  })
  .add('loading', () => {
    return (
      <UserInfoModal
        show={true}
        data={undefined}
        loading={true}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  })
  .add('graphlQL error', () => {
    return (
      <UserInfoModal
        show={true}
        data={{
          userInfo: { __typename: 'GenericErrorArray', errors: [{ message: 'GraphQL Error' }] },
        }}
        loading={false}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  })
  .add('network error', () => {
    return (
      <UserInfoModal
        show={true}
        data={null}
        loading={false}
        error={new ApolloError({ networkError: new Error('Network error') })}
        onClose={action('close modal')}
      />
    );
  })
  .add('custom', () => {
    const customData = {
      userInfo: {
        __typename: 'PublicUser',
        id: '1',
        avatarUrl: text('Avatar URL', `${appConfig.apiBasePath}/ava/m1.svg`),
        username: text('Username:', 'AuYHKS'),
        totalWager: number('Total Wager:', 0),
        totalProfit: number('Total Profit:', 0),
        mostPlayed: select(
          'Most played',
          {
            Dice: 'DICE',
            Goals: 'GOALS',
            Mines: 'MINES',
            Clams: 'CLAMS',
          },
          'CLAMS'
        ),
        totalBets: number('Total Bets:', 27),
        luckyBets: number('Won Bets:', 4),
      },
    };

    return (
      <UserInfoModal
        show={true}
        data={customData}
        loading={boolean('Loading', false)}
        error={undefined}
        onClose={action('close modal')}
      />
    );
  });
