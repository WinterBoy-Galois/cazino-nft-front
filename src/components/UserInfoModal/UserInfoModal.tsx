import React, { useCallback } from 'react';
import Modal from '../Modal';
import styles from './UserInfoModal.module.scss';
import { USER_INFO } from '../../graphql/queries';
import Username from '../Username';
import Button from '../Button';
import Loading from '../Loading';
import Error from '../Error';
import { ApolloError, useQuery } from '@apollo/client';
import DetailsContainer from '../DetailsContainer';
import { useLocation, useNavigate } from '@reach/router';
import UserStatistics from '../UserStatistics';

interface IProps {
  show: boolean;
  data: any;
  loading: boolean;
  error?: ApolloError;
  onClose?: () => void;
  onBack?: () => void;
}

const UserInfoModal: React.FC<IProps> = ({
  show,
  onClose,
  data,
  loading,
  error,
  onBack,
}: IProps) => {
  return (
    <Modal show={show} onClose={onClose} title="User Info">
      {loading ? <Loading /> : null}

      {error || data?.userInfo?.errors ? <Error>Could not load user info.</Error> : null}

      {data?.userInfo && !data?.userInfo?.errors && !loading ? (
        <div>
          <Username
            className={`${styles.username} ${styles['username--mobile']}`}
            username={data.userInfo.username}
            avatarUrl={data.userInfo.avatarUrl}
            loading={loading}
          />
          <DetailsContainer className={styles.details}>
            <Username
              className={`${styles.username} ${styles['username--desktop']}`}
              username={data.userInfo.username}
              avatarUrl={data.userInfo.avatarUrl}
              loading={loading}
            />

            <UserStatistics userStatistic={{ ...data.userInfo }} />
          </DetailsContainer>

          <div className={styles.button}>
            {onBack ? (
              <Button onClick={onBack}>Back</Button>
            ) : (
              <div className={styles.button__spacer} />
            )}
          </div>
        </div>
      ) : null}
    </Modal>
  );
};

interface IWithDataProps {
  show: boolean;
  userId: string;
  onClose?: () => void;
  backPath?: string;
  backState?: any;
}

const UserInfoModalWithData: React.FC<IWithDataProps> = ({
  show,
  userId,
  onClose,
  backPath,
  backState,
}: IWithDataProps) => {
  const { data, loading, error } = useQuery(USER_INFO, { variables: { userId } });
  const { pathname } = useLocation();
  const navigate = useNavigate();

  const handleBack = useCallback(
    () => backPath && backState && navigate(backPath, { state: backState }),
    [backPath, backState, navigate]
  );

  if (!userId && show) {
    navigate(pathname);
    return null;
  }

  return (
    <UserInfoModal
      show={show}
      data={data}
      loading={loading}
      error={error}
      onClose={onClose}
      onBack={backPath ? handleBack : undefined}
    />
  );
};

export default UserInfoModal;
export { UserInfoModalWithData };
