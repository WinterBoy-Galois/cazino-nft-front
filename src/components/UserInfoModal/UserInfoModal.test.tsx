import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { UserInfoModalWithData } from './UserInfoModal';
import { USER_INFO } from '../../graphql/queries';
import { GameTypes } from '../../models/gameTypes.model';
import { LocationProvider } from '@reach/router';

const mocks = [
  {
    request: {
      query: USER_INFO,
      variables: { userId: '1' },
    },
    result: {
      data: {
        userInfo: {
          __typename: 'PublicUser',
          id: '1',
          username: 'NIDHjQ',
          avatarUrl: 'https://test.de/ava/m1.svg',
          totalWager: 0,
          totalProfit: 0,
          mostPlayed: GameTypes.CLAMS,
          totalBets: 0,
          luckyBets: 0,
        },
      },
    },
  },
];

describe('LeaderboardsTab', () => {
  it('should match snapshot', async () => {
    // Arrange

    // Act
    const container = render(
      <LocationProvider>
        <MockedProvider mocks={mocks} addTypename={false}>
          <UserInfoModalWithData show={true} userId="1" />
        </MockedProvider>
      </LocationProvider>
    );

    // Assert
    expect(container).toMatchSnapshot();
  });
});
