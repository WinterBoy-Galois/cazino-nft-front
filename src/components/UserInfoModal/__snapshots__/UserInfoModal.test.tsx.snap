// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardsTab should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
        style="height: 768px;"
      >
        <div
          aria-modal="true"
          class="modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              User Info
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div
              class=" container"
            >
              <div
                class="spinner"
              >
                <span
                  class="spinner spinner--white "
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="height: 768px;"
    >
      <div
        aria-modal="true"
        class="modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            User Info
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div
            class=" container"
          >
            <div
              class="spinner"
            >
              <span
                class="spinner spinner--white "
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
